src/Util/SmartPaginatorFactory.php

```diff
@@ -102,24 +104,26 @@ private static function canDisableOutPutWalkers(QueryBuilder $queryBuilder): boo
         }
 
         // does not support queries using a field from a toMany relation in the ORDER BY clause
-        if (self::hasOrderByWithToManyAssociation($queryBuilder)) {
+        if (self::hasOrderByWithToManyAssociation($proxyQuery)) {
             return false;
         }
 
         return true;
     }
 
-    private static function hasOrderByWithToManyAssociation(QueryBuilder $queryBuilder): bool
+    private static function hasOrderByWithToManyAssociation(ProxyQueryInterface $proxyQuery): bool
     {
+        $queryBuilder = $proxyQuery->getQueryBuilder();
+
         $joinParts = $queryBuilder->getDQLPart('join');
 
         if (0 === \count($joinParts)) {
             return false;
         }
 
-        $orderByParts = $queryBuilder->getDQLPart('orderBy');
+        $sortBy = $proxyQuery->getSortBy();
```

Comment:
the sortBy for the paginator is only by one field right?

Reply:
I think you need to check both 
```
$queryBuilder->getDQLPart('orderBy');
```
and `getSortBy`.

Because people can call directly `addOrderBy` in the configureQuery.

Reply:
You're right, both things needs to be check:

```
        if ($this->getSortBy()) {
            $orderByDQLPart = $queryBuilder->getDQLPart('orderBy');
            $queryBuilder->resetDQLPart('orderBy');

            $sortBy = $this->getSortBy();
            if (false === strpos($sortBy, '.')) {
                $sortBy = $rootAlias.'.'.$sortBy;
            }

            $queryBuilder->addOrderBy($sortBy, $this->getSortOrder());
            foreach ($orderByDQLPart as $orderBy) {
                $queryBuilder->addOrderBy($orderBy);
            }
        }
```